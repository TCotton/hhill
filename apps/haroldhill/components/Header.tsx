import React from 'react'
import MenuItems from './MenuItems'
import {
  Button,
  Menu,
  MenuButton,
  MenuList,
  MenuItem,
  MenuItemOption,
  MenuGroup,
  MenuOptionGroup,
  MenuDivider,
} from '@chakra-ui/react'
import NavigationMenuDemo from './navigation'

const Header = () => {
  return (
    <header className="govuk-header" role="banner" data-module="govuk-header">
      <div className="govuk-header__container govuk-width-container">
        <div className="govuk-header__logo">
          <a
            href="#"
            className="govuk-header__link govuk-header__link--homepage">
            <span className="govuk-header__logotype">
              <span className="govuk-header__logotype-text">
                {/* eslint-disable-next-line react/no-unescaped-entities */}
                <em>HAROLD HILL</em>: A People's History
              </span>
            </span>
          </a>
        </div>
      </div>
      <div className="govuk-header__content govuk-width-container">
        <nav aria-label="Menu" className="govuk-header__navigation">
          <NavigationMenuDemo />
        </nav>
      </div>
    </header>
  )
}

export default Header
