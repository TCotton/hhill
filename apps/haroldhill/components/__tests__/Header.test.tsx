import { render, screen, waitFor } from '@testing-library/react'
import Header from '../Header'

jest.mock('isomorphic-unfetch', () =>
  jest.fn(() =>
    Promise.resolve({
      json: () =>
        Promise.resolve({
          message: 'ok',
          result: {
            items: [
              {
                fields: {
                  title: 'HAROLD HILL: A People’s History',
                  slug: '/',
                  pages: [
                    {
                      fields: {
                        title: 'Introduction',
                        contentRichText: 'string',
                        slug: 'introduction',
                        parent: 'HAROLD HILL: A People’s History',
                        id: 'string'
                      },
                      sys: {
                        id: 'StringID'
                      }
                    }
                  ]
                },
                sys: {
                  id: 'string'
                }
              }
            ]
          }
        })
    })
  )
)

describe('Header component', () => {
  it('renders correct href in HTML', () => {
    const { container } = render(<Header />)
    const link = container.querySelector('a')
    waitFor(() => {
      expect(link.getAttribute('href')).toBe('/')
    })
  })
  xit('should match snapshot', () => {
    const { baseElement } = render(<Header />)
    expect(baseElement).toMatchSnapshot()
  })
  it('should render the correct text', () => {
    render(<Header />)
    waitFor(() => {
      expect(screen.getByText(/HAROLD HILL/)).toBeInTheDocument()
      expect(screen.getByText(/A People's History/)).toBeInTheDocument()
      expect(screen.getByTestId('header-link')).toHaveAttribute('href', '/')
      expect(screen.getByTestId('header-link')).toHaveTextContent(
        "HAROLD HILL: A People's History"
      )
    })
  })
})
