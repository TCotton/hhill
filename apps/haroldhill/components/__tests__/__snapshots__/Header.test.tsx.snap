// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should match snapshot 1`] = `
<body>
  <div>
    <header
      class="govuk-header "
      data-module="govuk-header"
      role="banner"
    >
      <div
        class="govuk-header__container govuk-width-container"
      >
        <div
          class="govuk-grid-row"
        >
          <div
            class="govuk-grid-column-two-thirds-from-desktop center"
          >
            <p
              class="govuk-heading-m siteHeader"
            >
              <a
                class="govuk-link govuk-link--inverse govuk-link--no-underline header"
                data-testid="header-link"
                href="/"
              >
                <em>
                  HAROLD HILL
                </em>
                : A People's History
                 
              </a>
            </p>
          </div>
          <div
            class="govuk-grid-column-one-third-from-desktop"
          >
            <nav
              aria-label="Menu"
              class="govuk-header__navigation"
            />
          </div>
        </div>
      </div>
    </header>
  </div>
</body>
`;
